pkg/server/plugin/datastore/sql/mysql.go

```diff
@@ -18,23 +19,56 @@ import (
 	_ "github.com/spiffe/spire/pkg/server/plugin/datastore"
 )
 
-type mysql struct{}
+type mysqlDB struct{}
 
 const (
 	tlsConfigName = "spireCustomTLS"
 )
 
-func (my mysql) connect(cfg *configuration, isReadOnly bool) (*gorm.DB, error) {
+func (my mysqlDB) connect(cfg *configuration, isReadOnly bool) (*gorm.DB, string, bool, error) {
 	connString, err := configureConnection(cfg, isReadOnly)
 	if err != nil {
-		return nil, err
+		return nil, "", false, err
 	}
 
 	db, err := gorm.Open("mysql", connString)
 	if err != nil {
-		return nil, err
+		return nil, "", false, err
 	}
-	return db, nil
+
+	version, err := queryVersion(db, "SELECT VERSION()")
+	if err != nil {
+		return nil, "", false, err
+	}
+
+	supportsCTE, err := my.supportsCTE(db)
+	if err != nil {
+		return nil, "", false, err
+	}
+
+	return db, version, supportsCTE, nil
+}
+
+func (my mysqlDB) supportsCTE(gormDB *gorm.DB) (bool, error) {
+	db := gormDB.DB()
+	if db == nil {
+		return false, sqlError.New("unable to get raw database object")
```

Comment:
What kind of condition(s) could cause this? Will the error message(s) make sense to an operator? I imagine this would make it's way into a log somewhere down the stack

Reply:
This is more defensive. I don't think it would ever be a problem in practice. However, from GORMs docs:

> DB get `*sql.DB` from current connection If the underlying database connection is not a *sql.DB, returns nil